docs/Developers/Testing_coala_change_on_Bears.rst

```diff
@@ -0,0 +1,78 @@
+Testing changes to coala on coala-bears
+=======================================
+
+When large changes are made to the coala core, they should be tested
+on the coala-bears to confirm that they work as expected and don't
+break anything on the bears' side.
+
+There was an attempt to automate this through
+`#4424 <https://github.com/coala/coala/pull/4424>`_ which used to depend
+upon docker image, but unfortunately the image breaks stuff for now.
+
+But we can still do it manually, by changing the coala dependency for the
+bears to be the one consisting of the changes made. For this, we need to create
+a new branch on coala-bears, update the coala requirement to point to our
+changes and run it on CI.
+
+Set up a new testing branch
+---------------------------
+
+Assuming we have already made changes to coala through a new branch, we can
+use this branch to install coala instead of master by editing requirements.txt
+in coala-bears. Let's set-up a new branch on coala-bears for these
+changes.
+
+If you have not already, clone your fork by running:
+
+::
+
+    $ git clone https://github.com/<your-username>/coala-bears
+
+Make sure that you replace <your-username> with your Github username.
+
+Now, navigate to the directory where coala-bears is located.
+
+::
+
+    $ cd coala-bears
+
+Before making any change, switch to a new branch.
+
+::
+
+    $ git checkout -b new_test_branch
+
+.. note::
+    Never make any changes on your master branch.
+    See `Git Tutorial <https://coala.io/gitbasics>`_ for more info.
+
+Now that you're on a new branch, you are free to make changes.
+
+Change coala dependency in requirements.txt
+-------------------------------------------
+
+First, we need to change the coala dependency.
+
+Now, assuming you have made changes to coala fork on branch <branch-name>, you
+can edit the coala dependency by changing the corresponding line to:
+
+::
+
+    git+https://github.com/<your-username>/coala@<branch-name>#egg=coala
+
+Apply changes on coala-bears
+----------------------------
+
+Since the current changes aren't used anywhere on the bears, you need to
```

Comment:
Im not sure what "current changes" here are.

In bears the process should be:

1. edit requirements.txt to point to custom branch on fork
2. Create branch.  Suggest a branch name
3. Commit
4. Enable CI if not enabled
5. Push